### PR TITLE
Introduce inbuilt MsalCachePlugin to replace msal-node-extensions

### PR DESCRIPTION
**Pull request by cheenamalhotra**
It turns out msal-node-extensions is causing much pain than benefit. The latest insiders build fails to load extension with error:
`Cannot find module 'url-parse'` but fixing that, leads to below error (`dpapi.node not a valid Win32 application`):
image

This file 'dpapi.node' comes from the "msal-node-extensions" added for enabling cache persistence, which is adding dependency on running node-gyp, and since we don't do that in azurecore, we're having these issues. Sadly none of these issues were reproducible locally, due to this unexpected behavior:
See https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/3332

This PR implements MSAL cache plugin interface using 'Cryptr' to provide inbuilt solution for cache persistence, so we can remove dependency on msal-node-extensions.

I will wait for ad-hoc build to validate once again.

### CODE REVIEW

**Comment by cheenamalhotra** [comment]
Confirming I've validated the build from here: [Artifacts](https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=181357&view=artifacts&pathAsName=false&type=publishedArtifacts), and looks good on Windows.
@cssuh can you test this on macOS and confirm?

File: extensions/azurecore/src/account-provider/azureAccountProviderService.ts [hunk 1]
```diff
@@ -154,41 +154,25 @@ export class AzureAccountProviderService implements vscode.Disposable {
...
+				system: {
+					loggerOptions: {
+						loggerCallback: this.getLoggerCallback(),
+						logLevel: MsalLogLevel.Verbose,
+						piiLoggingEnabled: false,
+					},
 				},
 				cache: {
-					cachePlugin: persistenceCachePlugin
+					cachePlugin: this._cachePluginProvider?.getCachePlugin()
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/account-provider/azureAccountProviderService.ts:171]: I would suggest adding a comment here explaining why you're using our own custom one (basically what you put in the description)

Comment (cheenamalhotra) [reply extensions/azurecore/src/account-provider/azureAccountProviderService.ts:171]: Added here:
https://github.com/microsoft/azuredatastudio/pull/21335/files#diff-eb52caeb39b21d3d95a234f33f7f428ad4d42d60c6bdb6a5f839374811483d3cR90-R94

**Comment by cssuh** [comment]
Screenshot 2022-11-30 at 2 27 56 PM
I see this error when I try to open the app

**Comment by cheenamalhotra** [comment]
Ad-Hoc build is not signed, that could be the reason.

**Comment by cheenamalhotra** [comment]
@cssuh Thanks for the feedback, I've fixed the persistence issue so we should be good! 
I will proceed to merge it so Insiders build can pick up the changes and can follow up on anything remaining tomorrow.

**Pull request merged**